### ISSUE TITLE
Support for setOwner polyfill

### ISSUE THREAD

**Issue by offirgolan** [rwjblue/ember-getowner-polyfill#1]
To support multiple version of ember while dealing with container lookups and creating statically imported objects, there is a need for a `setOwner` polyfill.

For example:
```js
getOwner(this)._lookupFactory('validator:presence').create(options) // correctly sets up the owner/container
```

```js
import BaseValidator from '../validators/base';
BaseValidator.create() // Needs a way of setting the owner/container
```

**Comment by vitch** [comment]
I've just run into a related issue. Using `getOwner` via the polyfill on Ember 2.0.2 (in an addon which is consumed by multiple internal apps on different versions of Ember) I need to be able to create objects with a reference to the container.

[owner.ownerInjection](http://emberjs.com/api/classes/ContainerProxyMixin.html#method_ownerInjection) is the API to support this but it doesn't seem to be in the polyfill. I'll have a look into a possible PR...

**Comment by vitch** [comment]
See #3 for a potential fix for your case. With that branch you should be able to:

```javascript
import BaseValidator from '../validators/base';
BaseValidator.create(getOwner(this).ownerInjection);
```

**Comment by offirgolan** [comment]
Fantastic! Waiting on this to get merged and released :smile_cat:

**Comment by vitch** [comment]
Merged and `1.0.1` is released. Does this fix your problem?

**Comment by offirgolan** [comment]
I believe so. Thanks for your work! Feel free to close this issue, I'll
open a new one if I come across an issue.

On Wednesday, March 16, 2016, Kelvin Luck <notifications@github.com> wrote:

> Merged and 1.0.1 is released. Does this fix your problem?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/rwjblue/ember-getowner-polyfill/issues/1#issuecomment-197238983>
>


-- 
Thanks,
Offir Golan

**Comment by offirgolan** [comment]
Seems like all my tests are passing with the new `ownerInjection` polyfill (Ember 1.11 - canary). Thanks again for putting this together!
